app/services/digest_initiator_service.rb

```diff
@@ -8,6 +8,8 @@ def self.call(*args)
   end
 
   def call
+    digest_run = nil
```

Comment:
Rather than this being a distinct assignment can it be just returned from `run_with_advisory_lock` ?

Also what happens if lock is in progress, does this wait or isn't executed? just wondering if we need provisions for digest_run to be a nil value

Reply:
It blocks until it can get the lock and then runs. I've refactored this a bit and returned nil (which in turn bails out of the worker) if the `DigestRun` already exists.